// Copyright 2015 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
// License for the specific language governing permissions and limitations under
// the License.

syntax = "proto2";


import "row.proto";

package dsrf;

enum BlockType {
  HEAD = 0;
  FOOT = 1;
  BODY = 2;
}

message Block {

  // The block's type. We expect every file to contain one HEAD block, one
  // FOOT block and zero or more BODY blocks.
  optional BlockType type = 1;

  // If the block is a HEAD block, the report version should be mentioned.
  optional string version = 2;

  // The block number, as appear in the file (eg. "BL01" -> 1).
  optional int64 number = 3;

  // The file number in the report (eg. "3of4" -> 3).
  optional int32 file_number = 4;

  // The row objects in the block.
  repeated Row rows = 5;
}
